binding_generator.py

```diff
@@ -1453,15 +1453,23 @@ def generate_engine_class_source(class_api, used_classes, fully_used_classes, us
 
     if is_singleton:
         result.append(f"{class_name} *{class_name}::get_singleton() {{")
+        # We assume multi-threaded access is OK because each assignment will assign the same value every time
+        result.append(f"\tstatic {class_name} *singleton = nullptr;")
+        result.append("\tif (unlikely(singleton == nullptr)) {")
+        result.append(f"\t\tconst StringName _gde_class_name = {class_name}::get_class_static();")
         result.append(
-            f"\tstatic GDExtensionObjectPtr singleton_obj = internal::gdextension_interface_global_get_singleton({class_name}::get_class_static()._native_ptr());"
+            "\t\tGDExtensionObjectPtr singleton_obj = internal::gdextension_interface_global_get_singleton(_gde_class_name._native_ptr());"
```

Comment:
In PR #1176, we removed the intermediate `_gde_class_name` variable - this change adds it back!

Reply:
I guess it matters less here, because this only running the first time the singleton is retrieved :-/

But I think it'd still be nice to match the other cases where we're doing something similar

Reply:
It doesn't matter at all here... all that would do is to unnoticeably optimize a rare case of error. But I can do a second level of `if` if you really think we need this

Reply:
Yeah, I understand, it doesn't help optimize anything. At this point, it'd just be about consistency. We got rid of `_gde_class_name` everywhere else, and this brings it back in this one case.

> But I can do a second level of `if` if you really think we need this

I don't understand what the `if` would be for?

The change I'm imagining, is just turning this:

```cpp
        result.append(f"\t\tconst StringName _gde_class_name = {class_name}::get_class_static();")
        result.append(
            "\t\tGDExtensionObjectPtr singleton_obj = internal::gdextension_interface_global_get_singleton(_gde_class_name._native_ptr());"
        )
```

... into this:

```cpp
        result.append(
            f"\t\tGDExtensionObjectPtr singleton_obj = internal::gdextension_interface_global_get_singleton({class_name}::get_class_static()._native_ptr());"
        )
```

Reply:
Yeah I edited my message, there is no extra if needed